src/lib.rs

```diff
@@ -72,13 +84,23 @@ pub fn run_and_output(config: Config) {
 
 const MAX_FIX_ITERATIONS: usize = 10;
 
-fn run_per_file<'a>(
-    file_run_context: FileRunContext<'a>,
+fn run_per_file<
+    'a,
+    'b,
+    TFromFileRunContextInstanceProviderFactory: FromFileRunContextInstanceProviderFactory<Provider<'a> = TFromFileRunContextInstanceProvider>,
+    TFromFileRunContextInstanceProvider: FromFileRunContextInstanceProvider<'a, Parent = TFromFileRunContextInstanceProviderFactory>,
```

Comment:
I think I can remove this extra generic/bounds, this (per above comment) is what I tried to come up with that didn't work for describing the "parent-child" relationship, but so it got "superceded" by the other way (so this is "vestigial")